apisix/admin/upstreams.lua

```diff
@@ -33,17 +35,43 @@ local function check_conf(id, conf, need_id)
     return true
 end
 
+local function up_id_in_plugins(plugins, up_id)
+    if plugins and plugins["traffic-split"]
+        and plugins["traffic-split"].rules then
+
+        for _, rule in ipairs (plugins["traffic-split"].rules) do
+            local plugin_upstreams = rule.weighted_upstreams
+
+            for _, plugin_upstream in ipairs(plugin_upstreams) do
+                if plugin_upstream and plugin_upstream.upstream_id
+                    and tostring(plugin_upstream.upstream_id) == tostring(up_id) then
+
+                    return true
+                end
+            end
+        end
+
+        return false
+    end
+end
+
 
 local function delete_checker(id)
     local routes, routes_ver = get_routes()
     if routes_ver and routes then
         for _, route in ipairs(routes) do
-            if type(route) == "table" and route.value
-               and route.value.upstream_id
-               and tostring(route.value.upstream_id) == id then
-                return 400, {error_msg = "can not delete this upstream,"
-                                         .. " route [" .. route.value.id
-                                         .. "] is still using it now"}
+            if type(route) == "table" and route.value then
+                if up_id_in_plugins(route.value.plugins, id) == true then
+                    return 400, {error_msg = "can not delete this upstream,"
+                                             .. " traffic-split plugin in route [" .. route.value.id
```

Comment:
Better not to mention a particular plugin name, as we might add other plugin check later